lib/puppet/provider/mysql_datadir/mysql.rb

```diff
@@ -0,0 +1,65 @@
+require File.expand_path(File.join(File.dirname(__FILE__), '..', 'mysql'))
+Puppet::Type.type(:mysql_datadir).provide(:mysql, :parent => Puppet::Provider::Mysql) do
+
+  desc 'manage data directories for mysql instances'
+
+  commands :mysqld => 'mysqld'
+  commands :mysql_install_db => 'mysql_install_db'
+
+  def create
+    name                     = @resource[:name]
+    insecure                 = @resource.value(:insecure) || :true
```

Comment:
Please use just plain booleans here instead of `:true`/`:false`. Puppet core has even a separate boolean property: https://docs.puppetlabs.com/guides/custom_types.html#boolean-parameters